src/tests/ftest/util/server_utils_params.py

```diff
@@ -495,6 +494,7 @@ def __init__(self, base_namespace, index, provider=None, max_storage_tiers=MAX_S
             "ABT_ENV_MAX_NUM_XSTREAMS=100",
             "ABT_MAX_NUM_XSTREAMS=100",
             "DAOS_MD_CAP=1024",
+            "DAOS_POOL_RF=4",
```

Comment:
By moving this into `default_env_vars` the `DAOS_POOL_RF=4` will only be set for test that do NOT define a `/run/server_config/engines/x/env_vars` in their test yaml, or redfine it in that same yaml setting.  Just confirming that this is the intent.  For example, a test like, `src/tests/ftest/daos_test/rebuild.py` (because its yaml file defines specific `env_vars` entries - overriding the default) will no longer run with `DAOS_POOL_RF=4` with this change. There are currently ~50 functional tests that override the server config env_vars.

Reply:
@phender I am bit confused with current behavior. for this specific test, i want to reset DAOS_POOL_RF=2, but for all other tests, i want it to be default.

put DAOS_POOL_RF=4 in default_env_vars not match what i need, but don't move it, set it in the server yaml file won't change the value. would you point out how we should fix this? I asked this before...

Reply:
Which test do you want to use DAOS_POOL_RF=2?  If it is src/tests/ftest/control/dmg_pool_query_ranks.py, then we just need to include that entry in the test yaml (plus any other default values you'd like to retain).  Since the original src/tests/ftest/control/dmg_pool_query_ranks.yaml already defines a server_config/engines/x/env_vars entry it is already overriding the defaults that the test harness would otherwise include in the server yaml it generates. So we'd just need to add DAOS_POOL_RF=2:
```
server_config:
  name: daos_server
  crt_timeout: 5
  engines_per_host: 1
  engines:
    0:
      targets: 4
      nr_xs_helpers: 0
      env_vars:
        - SWIM_SUSPECT_TIMEOUT=10000
        - DAOS_POOL_RF=1
        - DD_MASK=io,epc,rebuild
        - DAOS_POOL_RF=2
```

Then the changes to src/tests/ftest/daos_test/suite.yaml and src/tests/ftest/util/server_utils_params.py can be reverted.

Reply:
I would probably prefer second method to avoid touching too many files.

Reply:
I've pushed an update that implements the second method.